graylog2-server/src/main/java/org/graylog2/streams/matchers/ExactMatcher.java

```diff
@@ -26,17 +26,13 @@
 
 	@Override
 	public boolean match(Message msg, StreamRule rule) {
-        if (!msg.hasField(rule.getField())) {
-            return false;
+        if (msg.getField(rule.getField()) == null) {
+            return rule.getInverted();
         }
 
-		Object value = msg.getField(rule.getField());
-		
-		if (value == null) {
-			return false;
-		}
-		
-		return rule.getInverted() ^ value.toString().trim().equals(rule.getValue());
+		final String value = msg.getField(rule.getField()).toString();
+
+		return rule.getInverted() ^ value.trim().equals(rule.getValue());
```

Comment:
Hm, I think I'm getting confused now. The `^` bitwise operator and the `||` logical operator also have different characteristics:

> For `^`, the result value is `true` if the operand values are different; otherwise, the result is `false`.

Is this the intended behavior? The `||` operator would return `true` if at least one of the operands is `true` while the `^` operator would return `true` only if the operands are different (e. g. both are `false`).